### ISSUE TITLE
Weird SELECT when table has the primary key

### ISSUE THREAD

**Issue by sayJason** [pingcap/tidb#33316]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
/* init */ CREATE TABLE t (c1 INT PRIMARY KEY, c2 INT);
/* init */ INSERT INTO t (c1, c2) VALUES (1, 1);

/* t1 */ BEGIN;
/* t1 */ UPDATE t SET c1=2, c2=2;
/* t2 */ BEGIN;
/* t2 */ DELETE FROM t;  -- t2 blocked
/* t1 */ COMMIT;  -- t2 unblocked
/* t2 */ SELECT * FROM t;
/* t2 */ COMMIT;
```

### 2. What did you expect to see? (Required)
The result of SELECT statement in the second transaction should get an empty set.
### 3. What did you see instead (Required)
The result of SELECT statement in the second transaction is (1, 1).

It is weird that SELECT statement is not empty only if table has the primary key.
### 4. What is your TiDB version? (Required)

```
+--------------------+
| version()          |
+--------------------+
| 5.7.25-TiDB-v5.4.0 |
+--------------------+
```

**Comment by morgo** [comment]
Closing as a duplicate of https://github.com/pingcap/tidb/issues/33315
